proc/catcher.go

```diff
@@ -0,0 +1,33 @@
+package proc
+
+import (
+	"github.com/brimdata/zed/zbuf"
+	"github.com/brimdata/zed/zqe"
+)
+
+type Catcher struct {
+	parent Interface
+}
+
+func NewCatcher(parent Interface) *Catcher {
+	return &Catcher{parent}
+}
+
+// SafePull runs a Pull and catches panics and turns them out errors.
+// This should be called out the output puller of a flowgraph and by
+// the top-level puller of all new goroutine created inside of a flowgraph.
```

Comment:
Move comment to Catcher and reword.
```suggestion
// Catcher wraps an Interface with a Pull method that recovers panics and turns them into errors.
// It should be wrapped around the output puller of a flowgraph and
// the top-level puller of any goroutine created inside of a flowgraph.
```